Userland/Utilities/passwd.cpp

```diff
@@ -89,6 +90,10 @@ int main(int argc, char** argv)
                 return 1;
             }
 
+            ScopeGuard current_password_zero_guard = [&] {
+                explicit_bzero(const_cast<char*>(current_password.value().characters()), current_password.value().length());
```

Comment:
This feels really gross. `StringImpl` contents are supposed to be immutable (and shareable).
Let's find a better way, perhaps we could have a simple buffer type that scrubs itself on destruction?

Reply:
Why not a flag in `StringImpl` that enables zeroing the StringImpl in the destructor?

Reply:
That solves one part of it, but `StringImpl` is still a shareable type, so one individual co-owner going out of scope doesn't mean the impl gets destroyed yet.
If someone copies the `String`, the `StringImpl` lives on.

Reply:
That's fine, because if someone copies the string, the data is in memory anyway.

Reply:
Ok, I created a `SecretString` class that wraps a `ByteBuffer`. It's much cleaner, let me know what you think.